src/Mailer.hs

```diff
@@ -0,0 +1,110 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Mailer where
+
+import Data.Maybe
+import Data.DateTime (getCurrentTime)
+import Data.Text (Text)
+import Data.Text.Format (format)
+import Data.UUID (UUID)
+import System.Environment (getEnv, lookupEnv)
+
+import qualified Network.HaskellNet.SMTP.SSL as SMTP
+import qualified Network.Mail.Mime as Mime
+
+import Registration (
+  EmailType(..), UserEvent(Emailed), UserState(..), Store, condenseConsecutive,
+  reactivelyRunAction, timeStampedAction)
+
+-- FIXME: we could extend this not to assume a port or using TLS (c.f STARTTLS)
+data MailerSettings = MailerSettings
+  { msSmtpSslSettings :: SMTP.Settings
+  , msServer :: String
+  , msUsername :: String
+  , msPassword :: String
+  , msSenderAddress :: Mime.Address
+  , msVerifiationLinkFormatter :: LinkFormatter
+  , msUnsubscribeLinkFormatter :: LinkFormatter
+  }
+
+instance Show MailerSettings where
+  show ms = "poop"
```

Comment:
Remove or say something about what you're showing?

Reply:
Oops. Yeah, I forgot about that. I declared it because I couldn't automatically defined one once I'd added the formatters. Do you think I should omit the password from the `Show` instance, so that one can never accidentally reveal it?

Reply:
Sounds like a good plan.